src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -984,7 +984,59 @@ private BoundExpression VisitRangePatternIndexerAccess(BoundImplicitIndexerAcces
             AddPlaceholderReplacement(node.ArgumentPlaceholders[1], rangeSizeExpr);
 
             var sliceCall = (BoundCall)node.IndexerOrSliceAccess;
-            var rewrittenIndexerAccess = VisitExpression(sliceCall);
+
+            // [start..] can be simplified to .Substring(start) or .Slice(start) for some built-in types
+            // e.g. string and (ReadOnly)Span
+
+            BoundExpression? moreEfficientIndexerAccess = null;
+            if (endMakeOffsetInput is null)
+            {
+                MethodSymbol? singleArgumentOverload = null;
+                NamedTypeSymbol? typeContainingElementType = null;
+
+                if (
+                    TryGetSpecialTypeMethod(node.Syntax, SpecialMember.System_String__Substring, out var stringSubstring, isOptional: true)
+                    && sliceCall.Method.Equals(stringSubstring)
+                )
+                {
+                    // If the overload is missing, we won't try the other `else if`
```

Comment:
Nit: this comment really doesn't say anything. Consider removing it.

Reply:
Removed